### PR TITLE
Expose Infinispan configuration in .Values #14

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

File: README.md.tpl [hunk 1]
```diff
@@ -37,3 +36,4 @@ Configure your {brandname} cluster by specifying values in the `deploy.*` sectio
 | `deploy.resourceLabels` | Adds labels to {brandname} resources such as pods and services. | `{}` | - |
 | `deploy.makeDataDirWritable` | Allows write access to the `data` directory for each {brandname} Server node. | false | Setting the value to `true` creates an initContainer that runs `chmod -R` on the `/opt/infinispan/server/data` directory and changes its permissions. |
 | `deploy.nameOverride` | Specifies a name for all {brandname} cluster resources. | Helm Chart release name | Configure a name for the created resources only if you need it to be different to the Helm Chart release name. |
+| `deploy.infinispan` | The {brandname} server configuration. | - | Care should be taken when modifying the default yaml to ensure that the Chart continues to function as expected. It is not recommended to modify the "metrics" property realm, endpoint or socket binding configuration. |
```
Comment (oraNod) [inline README.md.tpl:39]:
```suggestion
| `deploy.infinispan` | {brandname} Server configuration. | - | Modifies the underlying server instance of each pod. You should not change the default endpoints, socket bindings, or the security realm named "metrics". Modifying these default properties can result in unexpected behavior and loss of service.|
```

File: values.yaml [hunk 2]
```diff
@@ -47,3 +46,59 @@ deploy:
...
+      transport:
+        cluster: ${infinispan.cluster.name:cluster}
+        stack: kubernetes
+        node-name: ${infinispan.node.name:}
+    server:
+      interfaces:
+        - name: public
+          inetAddress:
+            value: ${infinispan.bind.address:127.0.0.1}
+      socketBindings:
```
Comment (oraNod) [inline values.yaml:63]:
```suggestion
            value: ${infinispan.bind.address:127.0.0.1}
      # Do not modify the default socket bindings.
      socketBindings:
```

File: values.yaml [hunk 3]
```diff
@@ -47,3 +46,59 @@ deploy:
...
+              clearText: secret
+        securityRealms:
+          - name: default
+            propertiesRealm:
+              groupsAttribute: Roles
+              userProperties:
+                path: users.properties
+              groupProperties:
+                path: groups.properties
+          - name: metrics
```
Comment (oraNod) [inline values.yaml:85]:
```suggestion
                path: groups.properties
          # Do not modify or remove the "metrics" security realm.
          - name: metrics
```

File: values.yaml [hunk 4]
```diff
@@ -47,3 +46,59 @@ deploy:
...
+            propertiesRealm:
+              groupsAttribute: Roles
+              userProperties:
+                path: metrics-users.properties
+                relativeTo: infinispan.server.config.path
+                plainText: true
+              groupProperties:
+                path: metrics-groups.properties
+                relativeTo: infinispan.server.config.path
+      endpoints:
```
Comment (oraNod) [inline values.yaml:95]:
```suggestion
                relativeTo: infinispan.server.config.path
      # Do not modify the default endpoints.
      endpoints:
```

**Review by oraNod — CHANGES_REQUESTED** [review]

**Comment by oraNod** [comment]
@ryanemerson not sure if you want to accept all those changes or not. should we add comments to the values.yaml directly?

File: README.md.tpl (hunk 1 above)
Comment (oraNod) [reply README.md.tpl:39]: @ryanemerson does "unexpected behaviour and loss of service" sound like too much?

File: values.yaml (hunk 4 above)
Comment (ryanemerson) [reply values.yaml:97]: I have moved this comment to the "metrics" endpoint and updated the message.

**Comment by ryanemerson** [comment]
> @ryanemerson not sure if you want to accept all those changes or not. should we add comments to the values.yaml directly?

Good thinking with the yaml comments. I have taken on board most of the suggestions. I have also added `[Prometheus]` to the comments to indicate why the config is required.

**Comment by ryanemerson** [comment]
Unfortunately this PR can't be merged until we have a solution for [ISPN-13261](You should not change the default endpoints, socket bindings, or the security realm named "metrics". Modifying these default properties can result in unexpected behavior and loss of service.).

Options:

1. Fix ISPN-13261
2. Make `.Values.deploy.Infinispan` an arbitrary Yaml string.

The main disadvantage of 2 is that malformed yaml is not recognised until pod creation. Approach 1 will prevent the chart being installed before any k8s resources are created.

**Review by oraNod — APPROVED** [review]

**Pull request merged**